src/core/Ident.ml

```diff
@@ -1,24 +1,21 @@
-type t = [`Anon | `Unqual of string | `Qual of string list * string | `Machine of string]
+type t = [`Anon | `User of string list * string | `Machine of string]
 
-let pp_modname modparts nm = List.fold_right (fun x acc -> x ^ "." ^ acc) modparts nm
+let pp_qual parts nm = List.fold_right (fun x acc -> x ^ "." ^ acc) parts nm
```

Comment:
Use `String.concat`.

Reply:
Another thing is that perhaps it's better for `pp_qual` to take `fmt` directly. That is,
```suggestion
let pp_qual fmt parts nm = Uuseg_string.pp_utf_8 fmt (String.concat "." @@ parts @ [nm])
```
(I did the coding on GitHub so maybe this would not compile...)

Reply:
Good point on the `concat`. We use this little helper in `to_string` as well, so I don't think we want to take in a formatter here.